stable/python/_kubeless.py

```diff
@@ -71,16 +74,13 @@ def handler():
         'event-type': req.get_header('event-type'),
         'event-time': req.get_header('event-time'),
         'event-namespace': req.get_header('event-namespace'),
-        'extensions': {
```

Comment:
The `req` object is not picklable. Therefore, it cannot be passed to a child process when using `spawn` or `forkserver`. 

This used to work with `fork` because nothing is actually pickled and sent to the child process when using `fork`. The child process used to share memory with the parent process and therefore had access to everything in the parent directly.

Reply:
I will have to check on that and get back to you. Will probably have to selectively discard some components in the `req` object which are not picklable (i.e, serializable).

Reply:
:+1: as far as I can tell, the most important thing is to keep request headers

Reply:
I have added a new class `PicklableBottleRequest` to convert the original request object into a request object that can be pickled (serialized). It does this by removing components that cannot be pickled.

**Components that are removed from the original request object:**
1. `wsgi.errors` - A file-like object that the Bottle app writes errors to after the request is complete. I don't see any use for this within the request itself.
2. `wsgi.input` - A file-like object that contains input data, forms, files, etc. This can be pickled as long as the length of the data is less than `MEMFILE_MAX` in which case Bottle uses in-memory buffers (`io.BytesIO`). If the length is greater than `MEMFILE_MAX`, Bottle uses `tempfile.TemporaryFile` which cannot be pickled. So, leaving this out altogether for now.
3. `bottle.app` - A reference to the Bottle application (`bottle.app()`) itself. I don't know why this would be useful.
4. `bottle.route` and `route.handle` - Both are reference to the `bottle.Route` object for the current request. In our case, we only have one route `/`. So, I don't see any use for this.

**Components that are retained:**
All custom headers and URL arguments (`?abc=X&def=Y`) along with all the convinience methods like `get_header()`. I have added an example of using these in `examples/request.py`.

Reply:
I am checking if it is possible to pickle the request body as well. If that can be done, the user code can access multipart form data as well.